MODCXMUX-30:Add support for codex-packages and codex-packages-sources methods

carolegodfrey opened the pull request:
## Purpose
Per https://issues.folio.org/browse/MODCXMUX-30 we are defining a common RAML definition for codex-packages and codex-packages-sources endpoints.

The following 3 methods are defined:

```
GET /codex-packages
GET /codex-packages/{id}
GET /codex-packages-sources
```
CQL for searching for Codex Packages is simpler than what is supported for Instances, for example

`name=academic and type=Print and ext.selected=true and source=kb sortby name`

Used package field mappings as define at:
https://docs.google.com/spreadsheets/d/1GQhgG5PnXJguB8K_JagmIVYpxx8HwGdkU4Sn96Q6YJg/edit#gid=1804402370

Used package filter information from input defined at:
https://docs.google.com/spreadsheets/d/1GQhgG5PnXJguB8K_JagmIVYpxx8HwGdkU4Sn96Q6YJg/edit#gid=1977344722
https://docs.google.com/spreadsheets/d/1GQhgG5PnXJguB8K_JagmIVYpxx8HwGdkU4Sn96Q6YJg/edit#gid=310763276

## Approach
- Used Codex instances RAML definition as a model for defining packages methods, parameters and object definitions.
- Created a separate definition file for packages methods. These methods will be part of a codex-packages interface that will be implemented by mod-codex-mux, mod-agreements and mod-codex-ekb
- Created codex_package_cqlschema.json defining commonly supported CQL for searching for packages.
- Created codex_package_cqlschema-ext.json defining extension for CQL for searching for selected packages.
- Executed `raml2html codex-packages.raml > codex-packages.html` to generate sample api document
- As test created temporary branch `add-codex-packages` in `mod-codex-ekb` project which references this branch for submodule.

## Screenshots
<img width="1669" alt="screen shot 2019-02-12 at 5 03 20 pm" src="https://user-images.githubusercontent.com/19415226/52671975-7748de80-2eea-11e9-8f2b-ae331a9f0121.png">
<img width="899" alt="screen shot 2019-02-12 at 5 03 54 pm" src="https://user-images.githubusercontent.com/19415226/52672362-a6ac1b00-2eeb-11e9-84f0-27cca3700c2b.png">

## Learning
- https://stackoverflow.com/questions/1777854/how-can-i-specify-a-branch-tag-when-adding-a-git-submodule
- https://dev.folio.org/guides/raml-cop/

carolegodfrey committed:
Added support for codex-packages and codex-pacakges-sources methods

id-jenkins commented:
<html><body><pre>
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 8 JSON schema files under directory './schemas'
ERROR: lint-raml-cop: CQLSchema.schema: Missing "description" for: capability, description, description, indexes, modifiers, name, name, name, operators, prefix, supported, type, type, uri, values, values
INFO: lint-raml-cop: tags.schema: Each property "description" is present.
ERROR: lint-raml-cop: errors.schema: Missing "description" for: total_records
INFO: lint-raml-cop: moduleInfo.schema: Each property "description" is present.
INFO: lint-raml-cop: metadata.schema: Each property "description" is present.
ERROR: lint-raml-cop: error.schema: Missing "description" for: code, message, parameters, type
ERROR: lint-raml-cop: resultInfo.schema: Missing "description" for: code, count, diagnostics, facetValues, facets, message, module, query, recordCount, responseTime, source, type, value
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls/codex
WARNING: lint-raml-cop: Missing from configuration: codex-packages.raml
WARNING: lint-raml-cop: Configuration needs to be updated (FOLIO-903).
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 12 JSON schema files under directory './schemas/codex'
INFO: lint-raml-cop: source.json: Each property "description" is present.
INFO: lint-raml-cop: instance.json: Each property "description" is present.
INFO: lint-raml-cop: packageCollection.json: Each property "description" is present.
INFO: lint-raml-cop: instanceCollection.json: Each property "description" is present.
INFO: lint-raml-cop: package.json: Each property "description" is present.
INFO: lint-raml-cop: sourceCollection.json: Each property "description" is present.
INFO: lint-raml-cop: coverage.json: Each property "description" is present.
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls/tagged-record-example
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 2 JSON schema files under directory './schemas/tagged-record-example'
INFO: lint-raml-cop: userdata.json: Each property "description" is present.
INFO: lint-raml-cop: userdataCollection.json: Each property "description" is present.
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
ERROR: lint-raml-cop: There were processing errors. See list above.
</pre><body></html>

dcrossleyau reviewed (commented):
I added configuration for this new codex-packages.raml file.

Wow, you added json schema descriptions too. Nice.
There is still the separate job for the other existing schema descriptions (not codex-related) which is why we are still seeing messages from the lint-raml-schema stage of CI.

Please do not re-add that old scripts/lint-raml-cop.sh script. We have the new one configured in CI. See the new one `folio-tools/lint-raml` via the linked doc at https://dev.folio.org/guides/raml-cop/ if needed to run locally.

I have not assessed any other technical aspects of this PR.

dcrossleyau committed:
Update copyright year FOLIO-1021

id-jenkins commented:
<html><body><pre>
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 8 JSON schema files under directory './schemas'
ERROR: lint-raml-cop: CQLSchema.schema: Missing "description" for: capability, description, description, indexes, modifiers, name, name, name, operators, prefix, supported, type, type, uri, values, values
INFO: lint-raml-cop: tags.schema: Each property "description" is present.
ERROR: lint-raml-cop: errors.schema: Missing "description" for: total_records
INFO: lint-raml-cop: moduleInfo.schema: Each property "description" is present.
INFO: lint-raml-cop: metadata.schema: Each property "description" is present.
ERROR: lint-raml-cop: error.schema: Missing "description" for: code, message, parameters, type
ERROR: lint-raml-cop: resultInfo.schema: Missing "description" for: code, count, diagnostics, facetValues, facets, message, module, query, recordCount, responseTime, source, type, value
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls/codex
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 12 JSON schema files under directory './schemas/codex'
INFO: lint-raml-cop: source.json: Each property "description" is present.
INFO: lint-raml-cop: instance.json: Each property "description" is present.
INFO: lint-raml-cop: packageCollection.json: Each property "description" is present.
INFO: lint-raml-cop: instanceCollection.json: Each property "description" is present.
INFO: lint-raml-cop: package.json: Each property "description" is present.
INFO: lint-raml-cop: sourceCollection.json: Each property "description" is present.
INFO: lint-raml-cop: coverage.json: Each property "description" is present.
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls/tagged-record-example
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 2 JSON schema files under directory './schemas/tagged-record-example'
INFO: lint-raml-cop: userdata.json: Each property "description" is present.
INFO: lint-raml-cop: userdataCollection.json: Each property "description" is present.
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
ERROR: lint-raml-cop: There were processing errors. See list above.
</pre><body></html>

carolegodfrey committed:
Correct raml cop errors and remove obsolete script

carolegodfrey committed:
Merge branch 'MODCXMUX-30-Codex-Package-Def' of https://github.com/folio-org/raml into MODCXMUX-30-Codex-Package-Def

id-jenkins commented:
<html><body><pre>
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 8 JSON schema files under directory './schemas'
ERROR: lint-raml-cop: Trouble loading CQLSchema.schema: Expecting property name enclosed in double quotes: line 10 column 5 (char 272)
INFO: lint-raml-cop: tags.schema: Each property "description" is present.
INFO: lint-raml-cop: errors.schema: Each property "description" is present.
INFO: lint-raml-cop: moduleInfo.schema: Each property "description" is present.
INFO: lint-raml-cop: metadata.schema: Each property "description" is present.
INFO: lint-raml-cop: error.schema: Each property "description" is present.
ERROR: lint-raml-cop: resultInfo.schema: Missing "description" for: diagnostics, type, value
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls/codex
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 12 JSON schema files under directory './schemas/codex'
INFO: lint-raml-cop: source.json: Each property "description" is present.
INFO: lint-raml-cop: instance.json: Each property "description" is present.
INFO: lint-raml-cop: packageCollection.json: Each property "description" is present.
INFO: lint-raml-cop: instanceCollection.json: Each property "description" is present.
INFO: lint-raml-cop: package.json: Each property "description" is present.
INFO: lint-raml-cop: sourceCollection.json: Each property "description" is present.
INFO: lint-raml-cop: coverage.json: Each property "description" is present.
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls/tagged-record-example
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 2 JSON schema files under directory './schemas/tagged-record-example'
INFO: lint-raml-cop: userdata.json: Each property "description" is present.
INFO: lint-raml-cop: userdataCollection.json: Each property "description" is present.
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
ERROR: lint-raml-cop: There were processing errors. See list above.
</pre><body></html>

carolegodfrey commented:
> I added configuration for this new codex-packages.raml file.
> 
> Wow, you added json schema descriptions too. Nice.
> There is still the separate job for the other existing schema descriptions (not codex-related) which is why we are still seeing messages from the lint-raml-schema stage of CI.
> 
> Please do not re-add that old scripts/lint-raml-cop.sh script. We have the new one configured in CI. See the new one `folio-tools/lint-raml` via the linked doc at https://dev.folio.org/guides/raml-cop/ if needed to run locally.
> 
> I have not assessed any other technical aspects of this PR.

Thanks for your help @dcrossleyau -- I removed the obsolete script and working through those other missing descriptions! :dizzy_face:

carolegodfrey committed:
Update descriptions for rmal cop errors

dcrossleyau commented:
Sorry, i must have confused you. Those other descriptions are some other job (FOLIO-1554), not this one. I was just explaining why Jenkins was reporting about them here.

But it is fantastic that you have done.

id-jenkins commented:
<html><body><pre>
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls
INFO: lint-raml-cop: Not assessing schema descriptions, as per option '--validate-only'.
INFO: lint-raml-cop: Assessing RAML files (https://dev.folio.org/guides/raml-cop/):
INFO: lint-raml-cop: Processing RAML v1.0 file: jsonSchemas.raml
INFO: lint-raml-cop:   raml-cop did not detect any errors with jsonSchemas.raml
INFO: lint-raml-cop: Processing RAML v1.0 file: ramls.raml
INFO: lint-raml-cop:   raml-cop did not detect any errors with ramls.raml
INFO: lint-raml-cop: Processing RAML v1.0 file: tenant.raml
INFO: lint-raml-cop:   raml-cop did not detect any errors with tenant.raml
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls/codex
INFO: lint-raml-cop: Not assessing schema descriptions, as per option '--validate-only'.
INFO: lint-raml-cop: Assessing RAML files (https://dev.folio.org/guides/raml-cop/):
INFO: lint-raml-cop: Processing RAML v1.0 file: codex.raml
ERROR: lint-raml-cop:   raml-cop detected errors with codex.raml:
[ramls/codex/codex.raml:11:2] WARNING Invalid JSON schema: Unexpected value 'Facet value'

INFO: lint-raml-cop: Processing RAML v1.0 file: codex-packages.raml
ERROR: lint-raml-cop:   raml-cop detected errors with codex-packages.raml:
[ramls/codex/codex-packages.raml:11:2] WARNING Invalid JSON schema: Unexpected value 'Facet value'

INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls/tagged-record-example
INFO: lint-raml-cop: Not assessing schema descriptions, as per option '--validate-only'.
INFO: lint-raml-cop: Assessing RAML files (https://dev.folio.org/guides/raml-cop/):
INFO: lint-raml-cop: Processing RAML v1.0 file: users.raml
ERROR: lint-raml-cop:   raml-cop detected errors with users.raml:
[ramls/tagged-record-example/users.raml:11:2] WARNING Invalid JSON schema: Unexpected value 'Facet value'

ERROR: lint-raml-cop: There were processing errors. See list above.
</pre><body></html>

id-jenkins commented:
<html><body><pre>
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 8 JSON schema files under directory './schemas'
INFO: lint-raml-cop: moduleInfo.schema: Each property "description" is present.
INFO: lint-raml-cop: errors.schema: Each property "description" is present.
INFO: lint-raml-cop: tags.schema: Each property "description" is present.
INFO: lint-raml-cop: metadata.schema: Each property "description" is present.
ERROR: lint-raml-cop: Trouble loading CQLSchema.schema: Expecting ',' delimiter: line 30 column 7 (char 782)
INFO: lint-raml-cop: error.schema: Each property "description" is present.
ERROR: lint-raml-cop: resultInfo.schema: Trouble determining "description" for property, perhaps misplaced.
ERROR: lint-raml-cop: resultInfo.schema: Missing "description" for: misplaced, value
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls/codex
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 12 JSON schema files under directory './schemas/codex'
INFO: lint-raml-cop: sourceCollection.json: Each property "description" is present.
INFO: lint-raml-cop: instance.json: Each property "description" is present.
INFO: lint-raml-cop: coverage.json: Each property "description" is present.
INFO: lint-raml-cop: packageCollection.json: Each property "description" is present.
INFO: lint-raml-cop: instanceCollection.json: Each property "description" is present.
INFO: lint-raml-cop: source.json: Each property "description" is present.
INFO: lint-raml-cop: package.json: Each property "description" is present.
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls/tagged-record-example
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 2 JSON schema files under directory './schemas/tagged-record-example'
INFO: lint-raml-cop: userdata.json: Each property "description" is present.
INFO: lint-raml-cop: userdataCollection.json: Each property "description" is present.
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
ERROR: lint-raml-cop: There were processing errors. See list above.
</pre><body></html>

carolegodfrey committed:
Update descriptions for rmal cop errors

id-jenkins commented:
<html><body><pre>
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 8 JSON schema files under directory './schemas'
ERROR: lint-raml-cop: Trouble loading CQLSchema.schema: Expecting ',' delimiter: line 30 column 7 (char 782)
INFO: lint-raml-cop: tags.schema: Each property "description" is present.
INFO: lint-raml-cop: errors.schema: Each property "description" is present.
INFO: lint-raml-cop: moduleInfo.schema: Each property "description" is present.
INFO: lint-raml-cop: metadata.schema: Each property "description" is present.
INFO: lint-raml-cop: error.schema: Each property "description" is present.
ERROR: lint-raml-cop: resultInfo.schema: Missing "description" for: value
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls/codex
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 12 JSON schema files under directory './schemas/codex'
INFO: lint-raml-cop: source.json: Each property "description" is present.
INFO: lint-raml-cop: instance.json: Each property "description" is present.
INFO: lint-raml-cop: packageCollection.json: Each property "description" is present.
INFO: lint-raml-cop: instanceCollection.json: Each property "description" is present.
INFO: lint-raml-cop: package.json: Each property "description" is present.
INFO: lint-raml-cop: sourceCollection.json: Each property "description" is present.
INFO: lint-raml-cop: coverage.json: Each property "description" is present.
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls/tagged-record-example
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 2 JSON schema files under directory './schemas/tagged-record-example'
INFO: lint-raml-cop: userdata.json: Each property "description" is present.
INFO: lint-raml-cop: userdataCollection.json: Each property "description" is present.
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
ERROR: lint-raml-cop: There were processing errors. See list above.
</pre><body></html>

carolegodfrey committed:
Update descriptions for rmal cop errors

id-jenkins commented:
<html><body><pre>
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 8 JSON schema files under directory './schemas'
ERROR: lint-raml-cop: Trouble loading CQLSchema.schema: Expecting ',' delimiter: line 51 column 13 (char 1319)
INFO: lint-raml-cop: tags.schema: Each property "description" is present.
INFO: lint-raml-cop: errors.schema: Each property "description" is present.
INFO: lint-raml-cop: moduleInfo.schema: Each property "description" is present.
INFO: lint-raml-cop: metadata.schema: Each property "description" is present.
INFO: lint-raml-cop: error.schema: Each property "description" is present.
ERROR: lint-raml-cop: resultInfo.schema: Missing "description" for: value
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls/codex
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 12 JSON schema files under directory './schemas/codex'
INFO: lint-raml-cop: source.json: Each property "description" is present.
INFO: lint-raml-cop: instance.json: Each property "description" is present.
INFO: lint-raml-cop: packageCollection.json: Each property "description" is present.
INFO: lint-raml-cop: instanceCollection.json: Each property "description" is present.
INFO: lint-raml-cop: package.json: Each property "description" is present.
INFO: lint-raml-cop: sourceCollection.json: Each property "description" is present.
INFO: lint-raml-cop: coverage.json: Each property "description" is present.
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls/tagged-record-example
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 2 JSON schema files under directory './schemas/tagged-record-example'
INFO: lint-raml-cop: userdata.json: Each property "description" is present.
INFO: lint-raml-cop: userdataCollection.json: Each property "description" is present.
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
ERROR: lint-raml-cop: There were processing errors. See list above.
</pre><body></html>

carolegodfrey committed:
Update descriptions for rmal cop errors

id-jenkins commented:
<html><body><pre>
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 8 JSON schema files under directory './schemas'
ERROR: lint-raml-cop: Trouble loading CQLSchema.schema: Expecting property name enclosed in double quotes: line 81 column 17 (char 2164)
INFO: lint-raml-cop: tags.schema: Each property "description" is present.
INFO: lint-raml-cop: errors.schema: Each property "description" is present.
INFO: lint-raml-cop: moduleInfo.schema: Each property "description" is present.
INFO: lint-raml-cop: metadata.schema: Each property "description" is present.
INFO: lint-raml-cop: error.schema: Each property "description" is present.
ERROR: lint-raml-cop: resultInfo.schema: Missing "description" for: value
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls/codex
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 12 JSON schema files under directory './schemas/codex'
INFO: lint-raml-cop: source.json: Each property "description" is present.
INFO: lint-raml-cop: instance.json: Each property "description" is present.
INFO: lint-raml-cop: packageCollection.json: Each property "description" is present.
INFO: lint-raml-cop: instanceCollection.json: Each property "description" is present.
INFO: lint-raml-cop: package.json: Each property "description" is present.
INFO: lint-raml-cop: sourceCollection.json: Each property "description" is present.
INFO: lint-raml-cop: coverage.json: Each property "description" is present.
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls/tagged-record-example
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 2 JSON schema files under directory './schemas/tagged-record-example'
INFO: lint-raml-cop: userdata.json: Each property "description" is present.
INFO: lint-raml-cop: userdataCollection.json: Each property "description" is present.
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
ERROR: lint-raml-cop: There were processing errors. See list above.
</pre><body></html>

carolegodfrey committed:
Update descriptions for rmal cop errors

id-jenkins commented:
<html><body><pre>
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 8 JSON schema files under directory './schemas'
ERROR: lint-raml-cop: CQLSchema.schema: Missing "description" for: name, type, uri
INFO: lint-raml-cop: tags.schema: Each property "description" is present.
INFO: lint-raml-cop: errors.schema: Each property "description" is present.
INFO: lint-raml-cop: moduleInfo.schema: Each property "description" is present.
INFO: lint-raml-cop: metadata.schema: Each property "description" is present.
INFO: lint-raml-cop: error.schema: Each property "description" is present.
ERROR: lint-raml-cop: resultInfo.schema: Missing "description" for: value
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls/codex
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 12 JSON schema files under directory './schemas/codex'
INFO: lint-raml-cop: source.json: Each property "description" is present.
INFO: lint-raml-cop: instance.json: Each property "description" is present.
INFO: lint-raml-cop: packageCollection.json: Each property "description" is present.
INFO: lint-raml-cop: instanceCollection.json: Each property "description" is present.
INFO: lint-raml-cop: package.json: Each property "description" is present.
INFO: lint-raml-cop: sourceCollection.json: Each property "description" is present.
INFO: lint-raml-cop: coverage.json: Each property "description" is present.
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls/tagged-record-example
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 2 JSON schema files under directory './schemas/tagged-record-example'
INFO: lint-raml-cop: userdata.json: Each property "description" is present.
INFO: lint-raml-cop: userdataCollection.json: Each property "description" is present.
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
ERROR: lint-raml-cop: There were processing errors. See list above.
</pre><body></html>

natali-zaitseva commented on schemas/codex/packageCollection.json:
```diff
@@ -0,0 +1,23 @@
+        "$ref": "package.json"
+      }
+    },
+    "resultInfo": {
```
seems that this section should look like 
`"resultInfo": {
  "description": "Result information schema",
  "$ref": "../resultInfo.schema"
    }`
this should fix jenkins  error `ERROR: lint-raml-cop: resultInfo.schema: Missing "description" for: value`

carolegodfrey replied:
Thanks @natali-zaitseva for talking a look and these details -- resolved these!

natali-zaitseva commented on schemas/CQLSchema.schema:
```diff
@@ -5,31 +5,37 @@
     "uri": {
-      "type": "string"
+      "type": "string",
+      "uri": "Uri of the property"
```
to fix jenkins `ERROR: lint-raml-cop: CQLSchema.schema: Missing "description" for: name, type, uri`
we probably should add 
`"description": "Uri of the property"`

natali-zaitseva commented on schemas/CQLSchema.schema:
```diff
@@ -5,31 +5,37 @@
+            "description": "Description of the Index"
           },
           "type": {
             "type": "string",
```
"description": "Type of the index"

natali-zaitseva commented on schemas/CQLSchema.schema:
```diff
@@ -5,31 +5,37 @@
         "type": "object",
         "properties": {
           "name": {
             "type": "string"
```
"description": "Name of the index"

carolegodfrey committed:
Added more descriptions

id-jenkins commented:
<html><body><pre>
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 8 JSON schema files under directory './schemas'
INFO: lint-raml-cop: CQLSchema.schema: Each property "description" is present.
INFO: lint-raml-cop: tags.schema: Each property "description" is present.
INFO: lint-raml-cop: errors.schema: Each property "description" is present.
INFO: lint-raml-cop: moduleInfo.schema: Each property "description" is present.
INFO: lint-raml-cop: metadata.schema: Each property "description" is present.
INFO: lint-raml-cop: error.schema: Each property "description" is present.
ERROR: lint-raml-cop: resultInfo.schema: Missing "description" for: value
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls/codex
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 12 JSON schema files under directory './schemas/codex'
INFO: lint-raml-cop: source.json: Each property "description" is present.
INFO: lint-raml-cop: instance.json: Each property "description" is present.
INFO: lint-raml-cop: packageCollection.json: Each property "description" is present.
INFO: lint-raml-cop: instanceCollection.json: Each property "description" is present.
INFO: lint-raml-cop: package.json: Each property "description" is present.
INFO: lint-raml-cop: sourceCollection.json: Each property "description" is present.
INFO: lint-raml-cop: coverage.json: Each property "description" is present.
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls/tagged-record-example
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 2 JSON schema files under directory './schemas/tagged-record-example'
INFO: lint-raml-cop: userdata.json: Each property "description" is present.
INFO: lint-raml-cop: userdataCollection.json: Each property "description" is present.
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
ERROR: lint-raml-cop: There were processing errors. See list above.
</pre><body></html>

carolegodfrey committed:
Revise selectionStatus to be an enum + more descriptions

id-jenkins commented:
<html><body><pre>
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 8 JSON schema files under directory './schemas'
INFO: lint-raml-cop: CQLSchema.schema: Each property "description" is present.
INFO: lint-raml-cop: tags.schema: Each property "description" is present.
INFO: lint-raml-cop: errors.schema: Each property "description" is present.
INFO: lint-raml-cop: moduleInfo.schema: Each property "description" is present.
INFO: lint-raml-cop: metadata.schema: Each property "description" is present.
INFO: lint-raml-cop: error.schema: Each property "description" is present.
ERROR: lint-raml-cop: resultInfo.schema: Missing "description" for: value
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls/codex
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 13 JSON schema files under directory './schemas/codex'
INFO: lint-raml-cop: source.json: Each property "description" is present.
INFO: lint-raml-cop: instance.json: Each property "description" is present.
INFO: lint-raml-cop: packageCollection.json: Each property "description" is present.
INFO: lint-raml-cop: instanceCollection.json: Each property "description" is present.
INFO: lint-raml-cop: package.json: Each property "description" is present.
INFO: lint-raml-cop: sourceCollection.json: Each property "description" is present.
INFO: lint-raml-cop: coverage.json: Each property "description" is present.
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
INFO: lint-raml-cop: Investigating and determining configuration: raml/ramls/tagged-record-example
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 2 JSON schema files under directory './schemas/tagged-record-example'
INFO: lint-raml-cop: userdata.json: Each property "description" is present.
INFO: lint-raml-cop: userdataCollection.json: Each property "description" is present.
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
ERROR: lint-raml-cop: There were processing errors. See list above.
</pre><body></html>

carolegodfrey committed:
Description on resultInfo schema value

carolegodfrey commented:
> Sorry, i must have confused you. Those other descriptions are some other job (FOLIO-1554), not this one. I was just explaining why Jenkins was reporting about them here.
> 
> But it is fantastic that you have done.

Just resolved the missing descriptions for all Codex related definitions (both new and existing)

dcrossleyau committed:
Fix text typo

dcrossleyau commented:
Hooray. Thanks for your great work.

dmtkachenko commented on schemas/codex/coverage.json:
```diff
@@ -0,0 +1,19 @@
+  "additionalProperties": false,
+  "properties": {
+    "beginCoverage": {
+      "type": "string",
```
shouldn't we also set 'format' as 'date' for beginCoverage and endCoverage fields
(see https://github.com/joelittlejohn/jsonschema2pojo/wiki/Reference#format)
.. potentially with 'customPattern'

carolegodfrey replied:
Thanks for recommendation @dmtkachenko -- added both format and customPattern properties

carolegodfrey committed:
Update coverage -- add format date and customPattern

carolegodfrey committed:
Merge branch 'MODCXMUX-30-Codex-Package-Def' of https://github.com/folio-org/raml into MODCXMUX-30-Codex-Package-Def

carolegodfrey committed:
Fix minor typo
